### ISSUE TITLE
Error: 'FunctionReference' isn't a type

### ISSUE THREAD

**Issue by pb-iatros** [dart-lang/dart_style#1051]
The upgrade from 2.0.3 to 2.1.0 throws an error when running build_runner

`$ flutter packages pub run build_runner build --delete-conflicting-outputs`

```
Failed to precompile build_runner:build_runner:
../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/dart_style-2.1.0/lib/src/source_visitor.dart:1809:31: Error: Type 'FunctionReference' not found.
  void visitFunctionReference(FunctionReference node) {
                              ^^^^^^^^^^^^^^^^^
../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/dart_style-2.1.0/lib/src/source_visitor.dart:1809:31: Error: 'FunctionReference' isn't a type.
  void visitFunctionReference(FunctionReference node) {
                              ^^^^^^^^^^^^^^^^^
```

**Comment by srawlins** [comment]
I think the analyzer dep needs to be bumped to `^1.6.0`:

https://pub.dev/packages/analyzer/changelog

If you run `pub upgrade` (or equivalent flutter command) and retry, does it still crash?

**Comment by pb-iatros** [comment]
We have a couple of circular dependencies that block the analyzer update, so we can't test that in the short term.
Meanwhile, we hope that someone is able to test your suggestion

**Comment by srawlins** [comment]
That's interesting. dart_style 2.1.0 requires analyzer `^1.3.0`. If you have some dependency that blocks upgrading to analyzer 1.6.0, the only situation I can think of is a dependency which has specified an explicit upper bound on analyzer's version which excludes 1.6.0. (A constraint with caret, like `^1.x.0`, should allow `1.6.0` for any `x` <= 6.) Do you know which package is allowing analyzer 1.3.0 but blocking analyzer 1.6.0?

**Comment by dened** [comment]
We did nothing(didn't upgrade any package), but have the same problems on CI.

**Comment by dened** [comment]
Ok, I broke my local code, did 'flutter pub upgrade', and have transitive dev_dependencies: dart_style 2.1.0 now.

**Comment by dened** [comment]
Adding override in pubspec.yaml  works
```
dependency_overrides:
  dart_style: 2.0.3
```

**Comment by munificent** [comment]
OK, I've published 2.1.1 which removes the code that depends on the latest analyzer and then published 2.2.0 which re-adds that code and also has a correct analyzer dependency constraint. This should make pub do the right thing when users depend on dart_style. Sorry for the pain.
